wondrous-app/components/Settings/Members/index.tsx

```diff
@@ -214,14 +221,24 @@ function Members() {
     }
   };
 
+  const handleRoleFilterChange = useCallback((roleId: string) => {
+    setSelectedRoleIds((selectedRoleIds) => {
+      const isSelected = selectedRoleIds.includes(roleId);
+      if (isSelected) {
+        return selectedRoleIds.filter((role) => role !== roleId);
+      }
+      return [...selectedRoleIds, roleId];
+    });
+  }, []);
+
   const handleSearchQueryOnChange = useCallback((ev: React.ChangeEvent) => {
     const searchQuery = (ev.target as HTMLInputElement).value;
     setSearchQuery(searchQuery);
   }, []);
 
   const handleSearchMembers = useCallback(
-    debounce(async (searchQuery: string) => {
-      if (!searchQuery.length) {
+    debounce(async () => {
```

Comment:
rn even if we have the debounce on timeout it will trigger the search for each character in the input

I think a fix would be to move this out of useEffect and call it on input change. Check for e.g components/CreateCollaborationModal/Steps/AddTeamMembers

Reply:
done